static/styles/zulip.scss

```diff
@@ -546,6 +546,59 @@ strong {
     color: inherit;
 }
 
+// Copied .dropdown-menu from v2.3.2
+// + https://github.com/zulip/zulip/commit/7a3a3be7e547d3e8f0ed00820835104867f2433d
+// basic idea of this fix is to remove decorations from :hover and apply them only
+// on :focus.
+.dropdown-menu {
+    li {
+        a {
+            display: block;
+            padding: 3px 20px;
+            clear: both;
+            font-weight: normal;
+            line-height: 20px;
+            color: hsl(0, 0%, 20%);
+            white-space: nowrap;
+
+            &:hover {
+                text-decoration: none;
+            }
+
+            &:focus {
+                color: hsl(0, 0%, 100%);
+                text-decoration: none;
+                background-color: hsl(200, 100%, 38%);
+                background-image: linear-gradient(
+                    to bottom,
+                    hsl(200, 100%, 40%),
+                    hsl(200, 100%, 35%)
+                );
+                background-repeat: repeat-x;
+                filter: progid:dximagetransform.microsoft.gradient(startcolorstr='hsla(328, 100%, 50%, 0.8)', endcolorstr='hsla(332, 100%, 50%, 0.7)', gradienttype=0);
+            }
+        }
+    }
+
+    .active > a {
+        color: hsl(0, 0%, 100%);
+        text-decoration: none;
+        background-color: hsl(200, 100%, 38%);
+        background-image: linear-gradient(
+            to bottom,
+            hsl(200, 100%, 40%),
+            hsl(200, 100%, 35%)
+        );
+        background-repeat: repeat-x;
+        outline: 0;
+        filter: progid:dximagetransform.microsoft.gradient(startcolorstr='hsla(328, 100%, 50%, 0.8)', endcolorstr='hsla(332, 100%, 50%, 0.7)', gradienttype=0);
+    }
+
+    .active > a:hover {
+        text-decoration: none;
+    }
+}
```

Comment:
I suspect this block can be fixed with this:
https://github.com/twbs/bootstrap/issues/7392#issuecomment-484513233

I think that's likely to be easier to maintain when doing future upgrades, since it just overrides the parameters relevant to this issue?

Reply:
I left it as it is, since the fix is not valid in our scenario since changed the structure of how typeahead's html is rendered, so any fix by bootstrap is not going to fix it. We have to maintain a fix of our own for it.